### DISCUSSION TITLE
Integrate B2B in Catalyst

### DISCUSSION THREAD

**Discussion by JesseMaxwell** [bigcommerce/catalyst#1213] [Q&A] [answered]
Is it possible to integrate the B2B portal (currently in Cornerstone) into Catalyst? If so, what is the approach?

**Comment by bc-john** [comment, marked as answer]
Hey @JesseMaxwell - 

We’re actively working towards a reference implementation of Catalyst with full B2B Edition functionality pre-layered in. 

Currently we’ve got this [Next.js Commerce sample repo](https://github.com/bigcommerce/nextjs-commerce) with the hosted Buyer Portal and associated functionality pre-wired. If you fill out [this form](https://docs.google.com/forms/d/e/1FAIpQLSfPXwJC541ZSH5u9CLABDD8CXM\_ojNET1L\_qLSKwTkXQiCFqQ/viewform), we can get you access to some other docs that will be useful. 

Note that the Next.js Commerce reference was built prior to [open sourcing the Buyer Portal](https://github.com/bigcommerce/b2b-buyer-portal). You should be able to leverage the same setup, but sub in your self hosted buyer portal url.  

There’s additional discussion [here](https://github.com/bigcommerce/catalyst/discussions/544) as well in case I missed something.
